python/tvm/relax/frontend/onnx_frontend.py

```diff
@@ -0,0 +1,1150 @@
+
+
+def _get_convert_map(opset):
+    return {
+        "MatMul": relay.frontend.onnx.MatMul,
+        "Concat": Concat,
+        "Add": Add,
+        "Mul": Mul,
+        "Cast": Cast,
+        "Gather": Gather,
+        "Gemm": Gemm,
+        "Reshape": relay.frontend.onnx.Reshape,
+        "Div": Div,
+        "Sigmoid": Sigmoid,
+        "Softmax": Softmax,
+        "Transpose": Transpose,
+        "Unsqueeze": Unsqueeze,
+        "Gelu": Gelu,
+        "BiasGelu": BiasGelu,
+        "Where": Where,
+        "Clip": Clip,
+        "Equal": Equal,
+        "Shape": Shape,
+        "Not": Not,
+        "Tanh": Tanh,
+        "Sqrt": Sqrt,
+        "Relu": Relu,
+        "Conv": Conv,
+        "Pow": Pow,
+        "Erf": Erf,
+        "CumSum": CumSum,
+        "Squeeze": Squeeze,
+        "Constant": Constant,
+        "Sub": Sub,
+        "LayerNormalization": relay.frontend.onnx.LayerNormalization,
+        "SkipLayerNormalization": relay.frontend.onnx.SkipLayerNormalization,
+        "EmbedLayerNormalization": relay.frontend.onnx.EmbedLayerNormalization,
+        # defs/reduction
+        "ReduceMax": relay.frontend.onnx.ReduceMax,
+        "ReduceMin": relay.frontend.onnx.ReduceMin,
+        "ReduceSum": relay.frontend.onnx.ReduceSum,
+        "ReduceMean": relay.frontend.onnx.ReduceMean,
+        "ReduceProd": relay.frontend.onnx.ReduceProd,
+        "ReduceLogSumExp": relay.frontend.onnx.ReduceLogSumExp,
+        "ReduceLogSum": relay.frontend.onnx.ReduceLogSum,
+        "ReduceSumSquare": relay.frontend.onnx.ReduceSumSquare,
+        "ReduceL1": relay.frontend.onnx.ReduceL1,
+        "ReduceL2": relay.frontend.onnx.ReduceL2,
+        "Expand": relay.frontend.onnx.Expand,
+        "ConstantOfShape": ConstantOfShape,
+        "Slice": Slice,
+        "Attention": relay.frontend.onnx.Attention,
+        "Pad": Pad,
+        "Split": Split,
+        "Tile": Tile,
+    }
+
+
+class GraphProto:
+    """A helper class for handling Relax expression copying from pb2.GraphProto.
+    Definition: https://github.com/onnx/onnx/blob/main/onnx/onnx.proto
+        Parameters
+    ----------
+    shape : dict of str to tuple, optional
+        The input shape to the graph
+    dtype : str or dict of str to str
+        The input types to the graph
+    """
+
+    current = None
+
+    def __init__(self, shape, dtype, target):
+        self._nodes = {}
+        self._inputs = {}
+        self._num_input = 0
+        self._shape = shape.copy() if shape else {}
+        self._input_names = []
+        self._dtype = dtype
+        self.opset = None
+        self._target = target
+        self.bb = relax.BlockBuilder()
+
+    def from_onnx(self, graph, opset) -> IRModule:
+        """Construct Relax expression from ONNX graph.
+        Onnx graph is a python protobuf object.
+        The companion parameters will be handled automatically.
+        However, the input names from onnx graph is vague, mixing inputs and
```

Comment:
some of this is user-facing API docs, could potentially go in the module-level docstring or in from_onnx

Reply:
Done